include/swift/AST/DiagnosticsParse.def

```diff
@@ -91,6 +91,9 @@ ERROR(forbidden_interpolated_string,none,
 ERROR(forbidden_extended_escaping_string,none,
       "%0 cannot be an extended escaping string literal", (StringRef))
 
+ERROR(regex_literal_parsing_error,none,
+      "%0", (StringRef))
+
```

Comment:
This is ok for now, but will drop source locations on the floor. We'll want to figure out a way to communicate source ranges back.

Reply:
We'll also want to be able to communicate back error codes, I think, so that diagnostics can be localized, customized, linked to educational notes, etc. And we'll want to be able to communicate multiple diagnostics, do fix-its, etc. But we should probably build all this up incrementally as we discover a need for it.